### PR TITLE
Scala version bump and minor addition to xsbti

### PR DESCRIPTION
**Pull request by eed3si9n**
/review @Duhemm, @dwijnand

### CODE REVIEW

**Comment by Duhemm** [comment]
`.travis.yml` still mentions Scala 2.10.5

File: internal/util-interface/src/main/java/xsbti/F1.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package xsbti;
+
+public interface F1<A1, R>
+{
+  R apply(A1 a1);
+}
```
Comment (dwijnand) [inline internal/util-interface/src/main/java/xsbti/F1.java:6]: No more java.util.List#add madness. Good call :+1:

**Comment by eed3si9n** [comment]
@Duhemm Nice catch on Travis

**Comment by eed3si9n** [comment]
I think I'm going to provide a concrete implementation for these interfaces instead of doing the hipster `new Intf` thing.

**Comment by dwijnand** [comment]
you might want to add static methods for constructing them as well, to remove the verbosity of creating them, for example:

``` java
  public static T2<A1, A2> of(A1 a1, A2 a2) {
    return new T2<A1, A2>(a1, a2);
  }
```

so that

``` java
T2<String, String> t2 = new T2<String, String>("foo", "bar");
```

becomes

``` java
T2<String, String> t2 = T2.of("foo", "bar");
```

**Comment by eed3si9n** [comment]
Added concrete classes in 277cbd12ef5a4e9ee42038c75f3265dbd2b27030.

File: internal/util-logging/src/main/scala/sbt/util/Logger.scala [hunk 2]
```diff
@@ -85,28 +85,12 @@ object Logger {
...
-      val sourcePath = o2m(sourcePath0)
-      val sourceFile = o2m(sourceFile0)
-    }
-
+  def f0[A](a: => A): F0[A] = InterfaceUtil.f0[A](a)
+  def m2o[A](m: Maybe[A]): Option[A] = InterfaceUtil.m2o(m)
+  def o2m[A](o: Option[A]): Maybe[A] = InterfaceUtil.o2m(o)
+  def position(line0: Option[Integer], content: String, offset0: Option[Integer], pointer0: Option[Integer],
+    pointerSpace0: Option[String], sourcePath0: Option[String], sourceFile0: Option[File]): Position =
+    InterfaceUtil.position(line0, content, offset0, pointer0, pointerSpace0, sourcePath0, sourceFile0)
```
Comment (dwijnand) [inline internal/util-logging/src/main/scala/sbt/util/Logger.scala:93]: Guess we could drop the 0's now.

File: internal/util-logging/src/main/scala/sbt/util/InterfaceUtil.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    val line = o2m(line0)
+    val lineContent = content
+    val offset = o2m(offset0)
+    val pointer = o2m(pointer0)
+    val pointerSpace = o2m(pointerSpace0)
+    val sourcePath = o2m(sourcePath0)
+    val sourceFile = o2m(sourceFile0)
+  }
+
+  private final class ConcreteProblem(
```
Comment (dwijnand) [inline internal/util-logging/src/main/scala/sbt/util/InterfaceUtil.scala:73]: We don't have anonymous inner problems now, we have concrete problems....

**Comment by dwijnand** [comment]
LGTM

**Pull request merged**